lib/delayed/worker.rb

```diff
@@ -247,7 +247,7 @@ def failed(job)
           say "Error when running failure callback: #{error}", 'error'
           say error.backtrace.join("\n"), 'error'
         ensure
-          self.class.destroy_failed_jobs ? job.destroy : job.fail!
+          destroy_failed_jobs(job) ? job.destroy : job.fail!
```

Comment:
Lets simplify this. Lets make the call `destroy_failed_jobs?` and just alway call `job.destroy_failed_jobs`.